lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1580,8 +1580,8 @@ def cmd_set(*args)
     # Set PAYLOAD
     if name.upcase == 'PAYLOAD' && active_module && (active_module.exploit? || active_module.evasion?)
       if value.start_with?('/', 'payload/')
```

Comment:
I think we can drop this guard clause now? 🤔 

```ruby
value = trim_path(value, "payload")

index_from_list(payload_show_results, value) do |mod|
    return false unless mod && mod.respond_to?(:first)

    # [name, class] from payload_show_results
    value = mod.first
end
```

Reply:
Indeed we can, good 👁!